src/main/java/uk/antiperson/stackmob/tasks/MergeTask.java

```diff
@@ -73,7 +73,7 @@ public void run() {
                 match.remove(false);
                 toRemove.add(match);
             }
-            if (size >= original.getMaxSize()) {
+            if (size + original.getSize() > original.getMaxSize()) {
                 for (int stackSize : Utilities.split(size, original.getMaxSize())) {
```

Comment:
Should this be size + original.getSize() rather than just size?

Reply:
Without this change, caused issues with max size reached during my tests.
This condition wanna ensure the entity can be incremented of "size", we must consider its previous size technically, because it's not necessary one. _(actually, incrementing the size if just a new set size)_

Perhaps it didn't occured on the old implementation because of its logic 🤷

Reply:
Are we talking about the same line? I was talking about line 77.

Reply:
Oops sorry 😅 I don't think either, an example would be enough I guess.

Assume there is an original entity (75) and max-size=100. Size is 43 and they want to merge into the original one.
We only need to duplicate the original of 43 since the original stays alive.